### PR TITLE
MAGECLOUD-2899: Redis Slave Nodes Have no Health Checks / Are not Stable

### PR DESCRIPTION
**Pull request by arhiopterecs**
https://magento2.atlassian.net/browse/MAGECLOUD-2899
### Description
In a case when using Redis slave, adds a couple of items to the cache settings in file `app/etc/env.php`. 
A feature of this configuration is that in case of failure Redis Slave, Magento should use master for reading.

### Fixed Issues (if relevant)
1. magento/ece-tools#MAGECLOUD-2899: Redis Slave Nodes Have no Health Checks / Are not Stable

### Manual testing scenarios
https://jira.corp.magento.com/browse/MAGECLOUD-48

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by michaelcasey316 — CHANGES_REQUESTED** [review]
According to Balázs Galántai, the integrity of the replica must be verified before using it in place of the primary (https://magento2.atlassian.net/browse/MAGECLOUD-2899?focusedCommentId=61967&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-61967). The example they gave involved the following commands:

`SENTINEL masters - Show a list of monitored masters and their state.`
`SENTINEL master <master name> -  Show the state and info of the specified master.`
`SENTINEL slaves <master name> - Show a list of slaves for this master, and their state.` 

Did you determine that integrity verification is unnecessary?

**Comment by kandy** [comment]
@michaelcasey316 Currently we do not need this functionality, because primary reason to use local instances is lower latency due to avoiding interaction over a network, and we verify that when slave fails we switch to master (see retry_reads_on_master option)

**Comment by michaelcasey316** [comment]
@kandy Not checking the state of the cluster before using the replica breaks strong consistency, doesn't it? Even if Magento is designed to never need up to date information in Redis, what about third party extensions that assume Redis doesn't have outdated data?

**Comment by kandy** [comment]
@michaelcasey316 Redis is not transaction storage and I don't think that is good  to expect consistency from it

**Review by michaelcasey316 — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

File: patches.json [hunk 1]
```diff
@@ -156,5 +156,12 @@
     "Fix monolog Slack Handler bug for magento 2.1.x": {
       "1.16.0": "MAGECLOUD-2793__fix_monolog_slack_handler_2.1.x.patch"
     }
+  },
+  "colinmollenhour/cache-backend-redis": {
```
Comment (YPyltiai) [inline patches.json]: Please re-define constraints to map to Magento Core versions.

**Review by bbatsche — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**